moveit_core/robot_state/src/robot_state.cpp

```diff
@@ -688,12 +688,8 @@ void moveit::core::RobotState::updateStateWithLinkAt(const LinkModel* link, cons
         if (cj[i] != child_link->getParentJointModel())
           updateLinkTransformsInternal(cj[i]);
     }
-    // update the root joint of the model to match (as best as possible given #DOF) the transform we wish to obtain for
-    // the root link.
-    // but I am disabling this code, since I do not think this function should modify variable values.
-    //    parent_link->getParentJointModel()->computeVariableValues(global_link_transforms_[parent_link->getLinkIndex()],
-    //                                                              position_ +
-    //                                                              parent_link->getParentJointModel()->getFirstVariableIndex());
+    // all collision body transforms are invalid now
+    dirty_collision_body_transforms_ = parent_link->getParentJointModel();
```

Comment:
i think this could adversely impact existing MoveIt! setups so we should not backport to Indigo

Reply:
This indeed changes behavior: It allows to update and subsequently use the warped collision transforms.
However, I don't see why this bug fix shouldn't be worth backporting.

Reply:
I agree that this causes more computational load when someone uses this (rather unusual) function with the `backward` parameter, but I don't see why we shouldn't backport it: without it the outdated transforms go unmarked.

Reply:
because it changes behavior on existing systems in ways we can't predict, and indigo is old and is suppose to be stable

Reply:
To clarify (please confirm @rhaschke) if you currently use `updateStateWithLinkAt(... backward=true)` in indigo, the code will use invalid transforms for collision checking (or it asserts and quits in debug build).
This is the only behavior these lines change by requiring an update of the invalid transforms.

Reply:
I don't use Indigo, and I didn't use `updateStateWithLinkAt(... backward=true)` either. I just discovered that the collision transforms are not (and cannot be) updated without this change.